autoload/test/javascript.vim

```diff
@@ -2,3 +2,9 @@ let test#javascript#patterns = {
   \ 'test': ['\v^\s*%(it|test)\s*[( ]\s*%("|'')(.*)%("|'')'],
   \ 'namespace': ['\v^\s*%(describe|suite|context)\s*[( ]\s*%("|'')(.*)%("|'')'],
 \}
+
+function! test#javascript#has_package(package) abort
+  exec "silent grep! '\b".a:package."\b' package.json"
+
+  return len(getqflist()) > 0
```

Comment:
I think it's better to use `readfile("package.json")` here, and I would also search for the package name surrounded by double quotes, just to make sure we don't detect another package which includes the test runner name as a substring.

```vim
for line in readfile("package.json")
  if line =~ '".a:package."'
    return 1
  endif
endfor
return 0
```